libbpf-rs/src/object.rs

```diff
@@ -86,48 +155,65 @@ impl Map {
         unimplemented!();
     }
 
-    /// Same as [`Map::lookup()`] except this also deletes the key from the map.
+    /// Same as [`LoadedMap::lookup()`] except this also deletes the key from the map.
     ///
     /// `key` must have exactly [`Map::key_size()`] elements.
-    pub fn lookup_and_delete(&mut self, _key: &[u8], _opts: MapOptions) -> Result<Option<Vec<u8>>> {
+    pub fn lookup_and_delete(
+        &mut self,
+        _key: &[u8],
+        _flags: LoadedMapFlags,
+    ) -> Result<Option<Vec<u8>>> {
         unimplemented!();
     }
 
     /// Update an element.
     ///
     /// `key` must have exactly [`Map::key_size()`] elements. `value` must have exatly
     /// [`Map::value_size()`] elements.
-    pub fn update(&mut self, _key: &[u8], _value: &[u8], _opts: MapOptions) -> Result<()> {
+    pub fn update(&mut self, _key: &[u8], _value: &[u8], _flags: LoadedMapFlags) -> Result<()> {
         unimplemented!();
     }
 }
 
-/// Options to configure [`Map`] operations.
-pub struct MapOptions {}
+/// Flags to configure [`Map`] operations.
+pub struct LoadedMapFlags {}
 
 /// Type of a [`Map`]. Maps to `enum bpf_map_type` in kernel uapi.
 #[non_exhaustive]
 pub enum MapType {}
 
-/// Represents a BPF program.
+/// Represents a parsed but not yet loaded BPF program.
 pub struct Program {}
 
 impl Program {
-    pub fn name(&self) -> &str {
+    pub fn prog_type(&mut self, _prog_type: ProgramType) -> &mut Self {
         unimplemented!();
     }
 
-    /// Name of the section this `Program` belongs to. This information is used by
-    /// [`crate::link::Link`] constructors to determine where to attach the prog.
-    pub fn section(&self) -> &str {
+    pub fn attach_type(&mut self, _attach_type: ProgramAttachType) -> &mut Self {
         unimplemented!();
     }
 
-    pub fn prog_type(&self) -> ProgramType {
+    pub fn ifindex(&mut self, _idx: i32) -> &mut Self {
         unimplemented!();
     }
 
-    pub fn attach_type(&self) -> ProgramAttachType {
+    pub fn allow_override(&mut self) -> &mut Self {
+        unimplemented!();
+    }
+
+    pub fn allow_multi(&mut self) -> &mut Self {
+        unimplemented!();
+    }
+
+    pub fn replace(&mut self) -> &mut Self {
+        unimplemented!();
+    }
```

Comment:
this is cgroup attach-specific flags, they are not properties of a program. These would be specified when doing bpf_program__attach_cgroup() in libbpf.

Reply:
ok